mim/commands/download.py

```diff
@@ -152,3 +187,59 @@ def download(package: str,
                     highlighted_error(f'{config_path} is not found.'))
 
     return checkpoints
+
+
+def _download_dataset(package: str, dataset: str, dest_root: str) -> None:
+    if platform.system() != 'Linux':
+        raise RuntimeError('download dataset is only available for Linux!')
+
+    if not is_installed(package):
+        raise RuntimeError(f'Please install {package} by ')
+
+    installed_path = get_installed_path(package)
+    mim_path = osp.join(installed_path, '.mim')
+    dataset_index_path = osp.join(mim_path, 'dataset-index.yml')
+
+    if not osp.exists(dataset_index_path):
+        raise FileNotFoundError(
+            f'Cannot find dataset-index.yaml in {dataset_index_path}, '
+            f'please update {package} to the latest version! If you have '
+            f'already updated it and still get this error, please report an '
+            f'issue to {package}')
+    with open(dataset_index_path) as f:
+        datasets_meta = yaml.load(f, Loader=yaml.SafeLoader)
+
+    if dataset not in datasets_meta:
+        raise KeyError(f'Cannot find {dataset} in {dataset_index_path}. '
+                       'here are the available datasets: '
+                       '{}'.format('\n'.join(datasets_meta.keys())))
+    dataset_meta = datasets_meta.get(dataset)
+
+    # TODO rename
+    script_path = dataset_meta.get('script')
+    script_path = osp.join(mim_path, script_path)
+    src_name = dataset_meta.get('dataset', dataset)
+    download_root = dataset_meta['download_root']
+    os.makedirs(download_root, exist_ok=True)
+
+    color_echo(f'Start downloading {dataset} to {download_root}...', 'blue')
+    process = subprocess.Popen(
+        ['odl', 'get', src_name, '-d', download_root],
+        stdin=sys.stdin,
+        stdout=subprocess.PIPE,
```

Comment:
I think `stdout=sys.stdout`, user can know it is downloading, how much has been downloaded, and how much time is needed.

Reply:
I initially redirected the output to `sys.stdout`, but the problem was that I couldn't capture the output of `odl` in the terminal and return a proper error message in `mim download`. Of course, this shouldn't have been the responsibility of `mim` to handle; `odl` should ensure the readability of its own error messages. Unfortunately, their error handling is not good, and they don't raise an error when there's no `login` but simply return "401: {"msg":"login required"} to stdout.

Regardless, redirecting the logs to a pipe indeed affected the user experience, so I have reverted the changes.